apps/dashboard/src/components/contract-components/hooks.ts

```diff
@@ -1,40 +1,31 @@
+"use client";
+
 import { useThirdwebClient } from "@/constants/thirdweb.client";
-import { getThirdwebClient } from "@/constants/thirdweb.server";
 import { queryOptions, useQuery } from "@tanstack/react-query";
 import type { Abi } from "abitype";
 import { isEnsName, resolveEns } from "lib/ens";
 import { useV5DashboardChain } from "lib/v5-adapter";
 import { useMemo } from "react";
 import { type ThirdwebContract, getContract } from "thirdweb";
-import {
-  resolveContractAbi,
-  fetchDeployMetadata as sdkFetchDeployMetadata,
-} from "thirdweb/contract";
-import {
-  getAllPublishedContracts,
-  getContractPublisher,
-  getPublishedUriFromCompilerUri,
-} from "thirdweb/extensions/thirdweb";
-import {
-  extractIPFSUri,
-  isAddress,
-  resolveImplementation,
-} from "thirdweb/utils";
-import invariant from "tiny-invariant";
+import { resolveContractAbi } from "thirdweb/contract";
+import { isAddress } from "thirdweb/utils";
 import {
   type PublishedContractWithVersion,
   fetchPublishedContractVersions,
   fetchPublisherProfile,
 } from "./fetch-contracts-with-versions";
+import { fetchDeployMetadata } from "./fetchDeployMetadata";
+import { fetchPublishedContracts } from "./fetchPublishedContracts";
+import { fetchPublishedContractsFromDeploy } from "./fetchPublishedContractsFromDeploy";
 import type { ContractId } from "./types";
 
 function isAnyObject(value: unknown): value is Record<string, unknown> {
   return typeof value === "object" && value !== null;
 }
 
-function removeUndefinedFromObjectDeep<T extends Record<string, unknown>>(
-  obj: T,
-): T {
+export function removeUndefinedFromObjectDeep<
```

Comment:
if we use this in other places probably worth extracting it into a util or something